libcloud/common/dimensiondata.py

```diff
@@ -732,7 +732,8 @@ class DimensionDataServerDisk(object):
     """
     A class that represents the disk on a server
     """
-    def __init__(self, id, scsi_id, size_gb, speed, state):
+    def __init__(self, id=None, scsi_id=None, size_gb=None, speed=None,
```

Comment:
Request for deploy server CaaS API https://<Cloud API URL>/caas/2.3/{org-id}/server/deployServer is requiring scsi_id and speed only. 

That's why I make them optional so that I could use this data class for deploy server. 

<deployServer xmlns="urn:didata.com:api:cloud:types"> 
      <name>Production FTPS Server</name>
      <description>This is the main FTPS Server</description> 
      <imageId>e0b48a16-f5c3-4045-bfd0-a1a7232a2710</imageId> 
      <start>true</start>      
      <administratorPassword>P$$ssWwrrdGoDd!</administratorPassword> 
      <cpu speed="HIGHPERFORMANCE" count="4" coresPerSocket="2"/>      
      <memoryGb>4</memoryGb>
      <network> <networkId>c550be0e-65c1-11e4-811f-005056806999</networkId>
      </network> 
      <primaryDns>10.20.255.12</primaryDns>    
      <secondaryDns>10.20.255.13</secondaryDns> 
      **<disk scsiId="0" speed="STANDARD"/>** 
      <microsoftTimeZone>040</microsoftTimeZone>
</deployServer>